src/Compilers/CSharp/Portable/Symbols/TypeUnification.cs

```diff
@@ -287,9 +287,10 @@ private static bool Contains(TypeSymbol type, TypeParameterSymbol typeParam)
                         NamedTypeSymbol namedType = (NamedTypeSymbol)type;
                         while ((object)namedType != null)
                         {
-                            foreach (TypeSymbol typeArg in namedType.TypeArgumentsNoUseSiteDiagnostics)
+                            ImmutableArray<TypeSymbol> typeParts = namedType.IsTupleType ? namedType.TupleElementTypes : namedType.TypeArgumentsNoUseSiteDiagnostics;
+                            foreach (TypeSymbol typePart in typeParts)
                             {
-                                if (Contains(typeArg, typeParam))
+                                if (Contains(typePart, typeParam))
```

Comment:
I didn't realize `VisitType` does not walk into error types. It probably should and it would be interesting to see what unit tests need to change as a result.

Given that difference, perhaps we should log a bug and update `VisitType` and `TypeUnification.Contains` in a separate PR.